notification-service/src/main/java/org/hypertrace/alert/engine/anomaly/event/processor/notification/WebhookNotifier.java

```diff
@@ -0,0 +1,59 @@
+package org.hypertrace.alert.engine.anomaly.event.processor.notification;
+
+import java.time.Instant;
+import org.hypertrace.alert.engine.anomaly.event.processor.NotificationChannel;
+import org.hypertrace.alert.engine.anomaly.event.processor.NotificationChannel.NotificationChannelConfig;
+import org.hypertrace.alert.engine.anomaly.event.processor.NotificationChannel.WebFormatNotificationChannelConfig;
+import org.hypertrace.alert.engine.anomaly.event.processor.NotificationChannelsReader;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.MetricAnomalyNotificationEvent;
+import org.hypertrace.alert.engine.notification.transport.webhook.WebhookSender;
+
+public class WebhookNotifier {
+
+  private final WebhookSender webhookSender;
+
+  public WebhookNotifier(WebhookSender webhookSender) {
+    this.webhookSender = webhookSender;
+  }
+
+  public void notify(
+      MetricAnomalyNotificationEvent metricAnomalyNotificationEvent,
+      NotificationChannel notificationChannel) {
+
+    if (!notificationChannel.getNotificationChannelConfig().isEmpty()) {
+
+      MetricAnomalyWebhookEvent message = convert(metricAnomalyNotificationEvent);
+      SlackMessage slackMessage = convert(message);
+      for (NotificationChannelConfig channelConfig :
+          notificationChannel.getNotificationChannelConfig()) {
+        if (!channelConfig
+            .getChannelConfigType()
+            .equals(NotificationChannelsReader.CHANNEL_CONFIG_TYPE_WEBHOOK)) {
+          continue;
+        }
+        WebFormatNotificationChannelConfig webFormatNotificationChannelConfig =
+            (WebFormatNotificationChannelConfig) channelConfig;
+        if (webFormatNotificationChannelConfig
+            .getWebhookFormat()
+            .equals(NotificationChannelsReader.WEBHOOK_FORMAT_SLACK)) {
+          webhookSender.send(webFormatNotificationChannelConfig.getUrl(), slackMessage);
```

Comment:
Followup: we need to discuss, what should be the action if notifiction sending has failed? Should we retry? But, this will be follow up.

Reply:
https://github.com/hypertrace/hypertrace/issues/267